src/twobinomial.jl

```diff
@@ -0,0 +1,541 @@
+#Score intervals for the difference of two binomial proportions
+@inline function mle_diff(p1::Float64, n1::Int, p2::Float64, n2::Int, δ::Float64)
+    if p1 - p2 - δ == 0 return 0.0 end
+    θ = n2 / n1
+    a = 1 + θ
+    b = -(1 + θ + p1 + θ * p2 + δ * (θ + 2))
+    c = δ^2 + δ * (2 * p1 + θ + 1) + p1 + θ * p2
+    d = -p1 * δ * (1 + δ)
+    v = (b / a / 3)^3 - b * c/(6 * a * a) + d / 2 / a
+    u = sign(v) * sqrt((b / 3 / a)^2 - c / 3 / a)
+    w = (pi + acos(v / u^3)) / 3
+    p1n = 2 * u * cos(w) - b / 3 /a
+    p2n = p1n - δ
+    return p1n, p2n
+end
+@inline function mn_diff_z_val(p1::Real, n1::Int, p2::Real, n2::Int, est::Real, δ::Real)
+    p1n, p2n = mle_diff(p1, n1, p2, n2, δ)
+    return (est - δ)^2 / ((n1 + n2) / (n1 + n2 - 1) * (p1n * (1 - p1n) / n1 + p2n * (1 - p2n) / n2))
+end
+function ci_diff_mn(test, level; atol::Float64 = 1E-8)
+    ests, ses, lcis, ucis = ci_diff_nhs_cc(test, level)
+    p1       = test.x1 / test.n1
+    p2       = test.x2 / test.n2
+    est      = p1 - p2
+    z        = quantile(Chisq(1), level)
+    fmnd(x)  = mn_diff_z_val(p1, test.n1, p2, test.n2, est, x) - z
+
+    if fmnd(lcis) * fmnd(est - eps()) < 0.0
+        ll = lcis
+        lu = est - eps()
+    else
+        ll = -1.0 + eps()
+        lu = lcis
+    end
+    if fmnd(ucis) * fmnd(est + eps()) < 0.0
+        ul = est + eps()
+        uu = ucis
+    else
+        ul = ucis
+        uu = 1.0 - eps()
+    end
+    lci = find_zero(fmnd, (ll, lu), atol = atol)
+    uci = find_zero(fmnd, (ul, uu), atol = atol)
+    return est, NaN, lci, uci
+end
+#Wald
+function ci_diff_wald(test, level)
+    alpha    = 1 - level
+    p1       = test.x1 / test.n1
+    p2       = test.x2 / test.n2
+    est      = p1 - p2
+    z        = quantile(Normal(), 1 - alpha / 2)
+    se       = sqrt(p1 * (1 - p1) / test.n1 + p2 * (1 - p2) / test.n2)
+    return est, se, est - z * se, est + z * se
+end
+#Wald continuity correction
+function ci_diff_wald_cc(test, level)
+    alpha    = 1 - level
+    p1       = test.x1 / test.n1
+    p2       = test.x2 / test.n2
+    est      = p1 - p2
+    z        = quantile(Normal(), 1 - alpha / 2)
+    se       = sqrt(p1 * (1 - p1) / test.n1 + p2 * (1 - p2) / test.n2)
+    cc       = 0.5*(1 / test.n1 + 1 / test.n2)
+    return est, se, est - z * se - cc, est + z * se + cc
+end
+#Newcombes Hybrid (wilson) Score interval for the difference of proportions
+#Newcombe 1998
+function ci_diff_nhs(test, level)
+    alpha    = 1 - level
+    p1       = test.x1 / test.n1
+    p2       = test.x2 / test.n2
+    est      = p1 - p2
+    z        = quantile(Normal(), 1 - alpha / 2)
+    est1, se1, lci1, uci1 = ci_prop_wilson(test.x1, test.n1, alpha)
+    est2, se2, lci2, uci2 = ci_prop_wilson(test.x2, test.n2, alpha)
+    return est, NaN, est - z * sqrt(lci1 * (1 - lci1)/test.n1 + uci2 * (1 - uci2) / test.n2), est + z * sqrt(uci1 * (1 - uci1) / test.n1 + lci2 * (1 - lci2) / test.n2)
```

Comment:
This line is hard to read. Can you store intermediate results in variables instead? Same in similar functions below.